src/proxy/admin_xa.go

```diff
@@ -0,0 +1,117 @@
+package proxy
+
+import (
+	"backend"
+	"fmt"
+	"router"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/xelabs/go-mysqlstack/sqlparser/depends/sqltypes"
+	"github.com/xelabs/go-mysqlstack/xlog"
+)
+
+const (
+	// It is more important to find the core cause to the prepared XA,
+	// so add thresholds to prevent operating too quickly.
+	intervalToNow = 10000 //one hour.
+)
+
+type AdminXA struct {
+	log     *xlog.Log
+	scatter *backend.Scatter
+	router  *router.Router
+	spanner *Spanner
+}
+
+func NewAdminXA(log *xlog.Log, scatter *backend.Scatter, router *router.Router, spanner *Spanner) *AdminXA {
+	return &AdminXA{
+		log:     log,
+		scatter: scatter,
+		router:  router,
+		spanner: spanner,
+	}
+}
+
+// Recover used to handle the 'XA RECOVER' to all backends .
+func (adminXA *AdminXA) Recover() (*sqltypes.Result, error) {
+	rewritten := fmt.Sprintf("XA RECOVER")
+	qr, err := adminXA.spanner.ExecuteSingle(rewritten)
+	if err != nil {
+		return nil, err
+	}
+	return qr, nil
+}
+
+// Commit used to handle the 'XA COMMIT' to all XAIDs an hour ago.
+func (adminXA *AdminXA) Commit() (*sqltypes.Result, error) {
+	log := adminXA.log
+	rewritten := fmt.Sprintf("XA RECOVER")
+	qr, err := adminXA.spanner.ExecuteSingle(rewritten)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, row := range qr.Rows {
+		// txn.xid = fmt.Sprintf("RXID-%v-%v", time.Now().Format("20060102150405"), txn.id)
```

Comment:
please remove the unused code

Reply:
Specially add the line,  let others know the format of the xaid.